### ISSUE TITLE
OpenGrok 0.7 does not allow searching across 'All' projects (Bugzilla #10758)

### ISSUE THREAD

**Issue by vladak** [oracle/opengrok#75]
status CLOSED severity _major_ in component _indexer_ for _---_
Reported in version _unspecified_ on platform _ANY/Generic_
Assigned to: Trond Norbye

On 2009-08-19 05:26:13 +0000, Harshit wrote:

> Hi,
> 
> As of OpenGrok 0.7 we need to select a Project before we can search in OpenGrok. But the main use case for which we use OpenGrok is to search for a term across 'All' projects.
> 
> Could you please help me know when this feature will be supported? Is it expected to come in the next release and when is the next release?
> 
> Thanks in advance.

On 2009-08-19 06:44:21 +0000, Lubos Kosco wrote:

> I'd like to close this as duplicate of http://defect.opensolaris.org/bz/show_bug.cgi?id=341 if you don't mind ...
> 
> we plan above bug in 0.8 , I checked newer lucene and it seems it has a support for this directly in its code, so I'd like to get lucene updated and then add a search through all the projects.
> Eventually I'd like to change the selector box in the way, that you can pick several projects and search only through them ...

On 2009-08-19 07:36:23 +0000, Harshit wrote:

> (In reply to comment # 1)
> 
> > I'd like to close this as duplicate of
> > http://defect.opensolaris.org/bz/show_bug.cgi?id=341 if you don't mind ...
> > 
> > we plan above bug in 0.8 , I checked newer lucene and it seems it has a support
> > for this directly in its code, so I'd like to get lucene updated and then add a
> > search through all the projects.
> > Eventually I'd like to change the selector box in the way, that you can pick
> > several projects and search only through them ...
> 
> Thanks Lubos..
> 
> Could you please help me know when you are planning to upgrade the Lucene? And when the OpenGrok 0.8 will be available? Knowing the dates will help us to plan. If this is going to be very far then we would like to move back to 0.6. Please help us know so that we can plan it accordingly.
> 
> Thanks for all the help.

On 2009-08-20 05:18:45 +0000, Harshit wrote:

> (In reply to comment # 1)
> 
> > we plan above bug in 0.8 , I checked newer lucene and it seems it has a support
> > for this directly in its code, so I'd like to get lucene updated and then add a
> > search through all the projects.
> 
> Hi Lubos/Trond, could you please help me know when you are planning to roll out OpenGrok 0.8 with upgraded Lucene?
> 
> > Eventually I'd like to change the selector box in the way, that you can pick
> > several projects and search only through them ...
> 
> A 'All' field in the selector would also be required along with the multi select selector box.

On 2009-08-21 09:19:12 +0000, Harshit wrote:

> (In reply to comment # 3)
> 
> > (In reply to comment # 1)
> > 
> > > we plan above bug in 0.8 , I checked newer lucene and it seems it has a support
> > > for this directly in its code, so I'd like to get lucene updated and then add a
> > > search through all the projects.
> > 
> > Hi Lubos/Trond, could you please help me know when you are planning to roll out
> > OpenGrok 0.8 with upgraded Lucene?
> > 
> > > Eventually I'd like to change the selector box in the way, that you can pick
> > > several projects and search only through them ...
> > 
> > A 'All' field in the selector would also be required along with the multi
> > select selector box.
> 
> Could you please reply to comment # 2 and comment # 3 ?
> 
> Thanks.

On 2009-08-21 15:16:30 +0000, Lubos Kosco wrote:

> (In reply to comment # 3)
> 
> > (In reply to comment # 1)
> > 
> > > we plan above bug in 0.8 , I checked newer lucene and it seems it has a support
> > > for this directly in its code, so I'd like to get lucene updated and then add a
> > > search through all the projects.
> > 
> > Hi Lubos/Trond, could you please help me know when you are planning to roll out
> > OpenGrok 0.8 with upgraded Lucene?
> 
> opengrok 0.8 will be out once db cache is ready + this bug is integrated
> I would be expecting (like) this to happen this autumn.
> 
> > > Eventually I'd like to change the selector box in the way, that you can pick
> > > several projects and search only through them ...
> > 
> > A 'All' field in the selector would also be required along with the multi
> > select selector box.
> 
> **\* This bug has been marked as a duplicate of bug 341 ***

On 2009-08-24 05:50:48 +0000, Harshit wrote:

> > opengrok 0.8 will be out once db cache is ready + this bug is integrated
> > I would be expecting (like) this to happen this autumn.
> 
> Hi Trond/Lubos,
> 
> Could you please help me know what all components need to be changed in the opengrok code? Just wanted to analyze the change required. If its simple and comprehendable, we might consider doing this.
> 
> Thanks.

On 2009-08-24 08:49:08 +0000, Lubos Kosco wrote:

> (In reply to comment # 6)
> 
> > > opengrok 0.8 will be out once db cache is ready + this bug is integrated
> > > I would be expecting (like) this to happen this autumn.
> > 
> > Hi Trond/Lubos,
> > 
> > Could you please help me know what all components need to be changed in the
> > opengrok code? Just wanted to analyze the change required. If its simple and
> 
> I am updating lucene code to latest this week, then we will have (besides 20% faster search):
> http://lucene.apache.org/java/2_4_1/api/org/apache/lucene/search/MultiSearcher.html
> and
> Parallel search available
> 
> the only change to code base then will be improving the jsp pages so they can pass on more projects (change to multiselect combobox) or a ticker for all projects, which will call above method ... quite simple, isn't it ? :)
> I hope I can do it within 2 weeks after lucene upgrade
> 
> the db cache is on Knuth and is the biggest priority currently, I haven't done any testing yet, so not sure how stable it is, but we need to sort out some details still.
> 
> > comprehendable, we might consider doing this.
> 
> well, if you know jsp/html, I'd be happy if someone can design the change to multiselect combobox in search and add a select all ticker
> I think a good idea is to put the project selector to the right side and add a ticker somewhere next to it ...
> 
> > Thanks.

On 2009-08-24 11:08:56 +0000, Harshit wrote:

> Thanks Lubos for the update.
> 
> Downgraded to 0.6 assuming that atleast it won't have the Projects selector. But it also has! Now moving down to 0.5 :(
> 
> Just a small question, is there any way, in 0.5, by which the configuration can be sent to webapp (the way we used to do in 0.7). I could not find where we are specifying the config url. Earlier it used to be a context-param in web.xml.
> 
> Could you please help me in this?
> 
> Thanks.

On 2009-08-24 11:36:06 +0000, Trond Norbye wrote:

> See EXAMPLE.txt :-)
> 
>  <context-param>
>    <param-name>ConfigAddress</param-name>
>    <param-value>localhost:2424</param-value>
>  </context-param>
